pyproject.toml

```diff
@@ -128,6 +129,7 @@ pytest-mock = "^3.12.0"
 ruff = "^0.3.0"
 black = "^24.2.0"
 pre-commit = "^3.7.0"
+ipykernel = "^6.29.4"
```

Comment:
is this addition needed? don't see it used in the changes

Reply:
Note needed, removed